backend/app/services/learning-service.py

```
@@ -0,0 +1,125 @@
+from typing import Dict, Optional, Any, Tuple
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)